src/services/milestones/createMilestoneSlug.js

```diff
@@ -0,0 +1,40 @@
+const async = require('async');
+const slugify = require('../../utils/slugify');
+
+/**
+ * This function checks if milestones name is unique in the campaign scope
+ * */
+const createMilestoneSlug = () => async context => {
```

Comment:
Let's create one createSlug function and use it on every project types (DAC, Milestone and Campaign). We can pass service/model to it.